### PR TITLE
[_]: feature/download-shared-folder

### PR DESCRIPTION
**Pull request by sraver**

### CODE REVIEW

File: src/app/routes/storage.js [hunk 1]
```diff
@@ -314,6 +314,24 @@ module.exports = (Router, Service, App) => {
       });
   });
 
+  Router.get('/storage/shared-folder/:token', async (req, res) => {
+    const result = await Service.Share.getFolderInfo(req.params.token);
+    res.status(200).json(result);
+  });
+
+  Router.post('/storage/share/down/folders', async (req, res) => {
```
Comment (sg-gs) [inline src/app/routes/storage.js:322]: This is a GET with query params that could be obtained through `req.params`

File: src/app/services/share.js [hunk 2]
```diff
@@ -1,33 +1,68 @@
...
-const FolderService = require('./folder');
-const lib = require('@internxt/lib');
+const { Environment } = require('@internxt/inxt-js');
+const { aes } = require('@internxt/lib');
 
 const { Op } = sequelize;
 
 module.exports = (Model, App) => {
-  const FolderServiceInstance = FolderService(Model, App);
+  const maxAcceptableSize = 1024 * 1024 * 1000; // 1000MB
```
Comment (sg-gs) [inline src/app/services/share.js:10]: If `maxAcceptableSize` is supposed to be 1GB, then use 1024x1024x**1024**

**Review by sg-gs — CHANGES_REQUESTED** [review]

File: src/app/services/share.js [hunk 3]
```diff
@@ -38,7 +73,215 @@ module.exports = (Model, App) => {
...
+    const { user_id } = await Model.users.findOne({
+      raw: true,
+      where: {
+        email: { [Op.eq]: email },
+      },
+      attributes: ['user_id']
+    });
+    return new Environment({
+      bridgePass: user_id,
+      bridgeUser: email,
```
[the remark this answers is not in the document]
Comment (sraver) [reply src/app/services/share.js:245]: you mean that I should get `getTeamByEmail(email)` and then use the `bridge_user` that comes from there? @sg-gs

File: src/app/routes/storage.js [hunk 4]
```diff
@@ -314,6 +314,24 @@ module.exports = (Router, Service, App) => {
...
+    let { token, directoryId, offset, limit } = req.body;
+    const result = await Service.Share.getDirectoryFolders(directoryId, offset, limit, token);
+    res.status(200).json(result);
+  });
+
+  Router.post('/storage/share/down/files', async (req, res) => {
```
[the remark this answers is not in the document]
Comment (sraver) [reply src/app/routes/storage.js:328]: You prefer to send the four parameters on the query? And I guess to do the same on the request of folders right? @sg-gs

File: src/app/services/share.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (sg-gs) [reply src/app/services/share.js:245]: I am referring that the value that should be used as `bridgeUser` is the field bridge_user from user when is queried in L236:

```
const { user_id, **bridge_user** } = await Model.users.findOne({
    raw: true,
    where: {
      email: { [Op.eq]: email },
    },
    attributes: ['user_id']
  });
```

**Review by sg-gs — APPROVED** [review]

**Pull request merged**